MPF.Processors/Redumper.cs

```diff
@@ -219,75 +223,47 @@ public override void GenerateSubmissionInfo(SubmissionInfo info, MediaType? medi
-                        info.CommonDiscInfo.Serial = xmid.Serial ?? string.Empty;
+                        info.CommonDiscInfo!.CommentsSpecialFields![SiteCode.XMID] = xmidString;
+                        var xmid = SabreTools.Serialization.Wrappers.XMID.Create(xmidString);
+                        info.CommonDiscInfo.Serial = xmid?.Serial ?? string.Empty;
                         if (!redumpCompat)
                         {
-                            info.VersionAndEditions!.Version = xmid.Version ?? string.Empty;
-                            info.CommonDiscInfo.Region = ProcessingTool.GetXGDRegion(xmid.Model.RegionIdentifier);
+                            info.VersionAndEditions!.Version = xmid?.Version ?? string.Empty;
+                            info.CommonDiscInfo.Region = ProcessingTool.GetXGDRegion(xmid?.Model.RegionIdentifier);
                         }
                     }
-
-                    string? dmi1Crc = HashTool.GetFileHash($"{basePath}.dmi", HashType.CRC32);
-                    if (dmi1Crc != null)
-                        info.CommonDiscInfo!.CommentsSpecialFields![SiteCode.DMIHash] = dmi1Crc.ToUpperInvariant();
-                    string? pfi1Crc = HashTool.GetFileHash($"{basePath}.pfi", HashType.CRC32);
-                    if (pfi1Crc != null)
-                        info.CommonDiscInfo!.CommentsSpecialFields![SiteCode.PFIHash] = pfi1Crc.ToUpperInvariant();
-                    if (ProcessingTool.IsValidSS($"{basePath}.ss"))
-                    {
-                        string? ss1Crc = HashTool.GetFileHash($"{basePath}.ss", HashType.CRC32);
-                        if (ss1Crc != null)
-                            info.CommonDiscInfo!.CommentsSpecialFields![SiteCode.SSHash] = ss1Crc.ToUpperInvariant();
-                    }
-
-                    string? ranges1 = ProcessingTool.GetSSRanges($"{basePath}.ss");
-                    if (!string.IsNullOrEmpty(ranges1))
-                        info.Extras!.SecuritySectorRanges = ranges1;
-
-                    break;
-
-                case RedumpSystem.MicrosoftXbox360:
-                    // If .dmi / .pfi / .ss don't already exist, create them
-                    if (!File.Exists($"{basePath}.dmi"))
-                        RemoveHeaderAndTrim($"{basePath}.manufacturer", $"{basePath}.dmi");
-                    if (!File.Exists($"{basePath}.pfi"))
-                        RemoveHeaderAndTrim($"{basePath}.physical", $"{basePath}.pfi");
-                    if (!File.Exists($"{basePath}.ss"))
-                        ProcessingTool.CleanSS($"{basePath}.security", $"{basePath}.ss");
-
-                    string xemidString = ProcessingTool.GetXeMID($"{basePath}.dmi");
-                    var xemid = SabreTools.Serialization.Wrappers.XeMID.Create(xemidString);
-                    if (xemid != null)
+                    string xemidString = ProcessingTool.GetXeMID($"{basePath}.dmi").Trim('\0');
```

Comment:
Extra newline needed above this line

Reply:
Wrong line. Above `string xemidString = ProcessingTool.GetXeMID($"{basePath}.dmi").Trim('\0');`